List permissions fix No. 3 in a series

ckolos opened the pull request:
So maybe there are two ways to create a list in shavar and maybe we use
one not the other. Let's be sure and set object acl's either way.

I blame someone else for this not being fixed the first time.

skim1102 commented on publish2cloud.py:
```diff
@@ -205,7 +206,8 @@ def publish_to_s3(config, section, chunknum):
         k.key = key_name
-        k.set_contents_from_filename(output_filename, policy='bucket-owner-full-control')
+        k.set_contents_from_filename(output_filename)
+        k.set_acl('bucket-owner-full-control')
```
```suggestion
        k.set_acl('public-read')
```
According to the documentation [here](http://boto.cloudhackers.com/en/latest/s3_tut.html#setting-getting-the-access-control-list-for-buckets-and-keys), if we want to have the files be publicly readable the policy should be `public-read`.

skim1102 commented on publish2cloud.py:
```diff
@@ -154,6 +154,7 @@ def new_data_to_publish_to_s3(config, section, new):
             key.set_contents_from_string('a:1:32:32\n' + 32 * '1')
         current = tempfile.TemporaryFile()
         key.get_contents_to_file(current)
+        key.set_acl('bucket-owner-full-control')
```
```suggestion
        key.set_acl('public-read')
```

ckolos-moz-admin commented:
After chatting it out with our security folk and AWS support, no one could state that it was a terrible idea to have the cloudfront object access id in public, but no one would also commit to it not being a terrible idea.

I've switched the backend job for shavar list generation to pass in the necessary credentials and tested list generation both from my local host as well as from jenkins. All works as expected. I:

- Deleted the digest256 file
- deleted the chunk file
- re-ran list-generation to recreate the file
- verified that the permissions on the re-created files now included the cloudfront distro's access
- verified that I could pull down the resources from cloudfront using the form: <CDN Url>/<key>

I then re-set the permissions on all existing files to allow cloudfront read and tested *all* files currently in stage.

I think this works.

ckolos commented:
After chatting it out with our security folk and AWS support, no one could state that it was a terrible idea to have the cloudfront object access id in public, but no one would also commit to it not being a terrible idea either.

I've switched the backend job for shavar list generation to pass in the necessary credentials and tested list generation both from my local host as well as from jenkins. All works as expected. I:

- Deleted the digest256 file
- deleted the chunk file
- re-ran list-generation to recreate the file
- verified that the permissions on the re-created files now included the cloudfront distro's access
- verified that I could pull down the resources from cloudfront using the form: <CDN Url>/<key>

I then re-set the permissions on all existing files to allow cloudfront read and tested *all* files currently in stage.

I think this works.
